doc/network-stack.md

```diff
@@ -0,0 +1,1146 @@
+
+4. As the listeners for port 80 and port 443 are existing, no changes for listeners.
+
+### Push Another App
+
+No changes to Envoy config of existing app(s). No direct app-to-app communication is possible as of now.
+
+
+### Map Additional Route
+Map a new route to the existing app test-node-app:
+
+```cf map-route test-node-app cf.cfi759.istio.shoot.canary.k8s-hana.ondemand.com --hostname my-app```
+
+#### Changes on Istio and cf-for-k8s components
+
+The CloudController creates a new `Route CR`. This is a representation of the cf route. It contains route_guid and a list of destinations:
+
+```json
+$ kubectl get route -n cf-workloads 9fa832fa-4054-430f-9fc4-6d82733df836 -o json                 
+
+{
+    "apiVersion": "networking.cloudfoundry.org/v1alpha1",
+    "kind": "Route",
+    "metadata": {
+        "creationTimestamp": "2020-07-01T10:32:58Z",
+        "finalizers": [
+            "routes.networking.cloudfoundry.org"
+        ],
+        "generation": 3,
+        "labels": {
+            "app.kubernetes.io/component": "cf-networking",
+            "app.kubernetes.io/managed-by": "cloudfoundry",
+            "app.kubernetes.io/name": "9fa832fa-4054-430f-9fc4-6d82733df836",
+            "app.kubernetes.io/part-of": "cloudfoundry",
+            "app.kubernetes.io/version": "0.0.0",
+            "cloudfoundry.org/domain_guid": "48668620-1f88-45ca-a189-c96eda6972aa",
+            "cloudfoundry.org/org_guid": "feb7b668-e649-4d52-b973-ea758ae4cff1",
+            "cloudfoundry.org/route_guid": "9fa832fa-4054-430f-9fc4-6d82733df836",
+            "cloudfoundry.org/space_guid": "159b4307-970a-419e-a6ba-962838a7cb4a"
+        },
+        "name": "9fa832fa-4054-430f-9fc4-6d82733df836",
+        "namespace": "cf-workloads",
+        "resourceVersion": "125809",
+        "selfLink": "/apis/networking.cloudfoundry.org/v1alpha1/namespaces/cf-workloads/routes/9fa832fa-4054-430f-9fc4-6d82733df836",
+        "uid": "e56b639a-de86-4aba-b6df-af4550348447"
+    },
+    "spec": {
+        "destinations": [
+            {
+                "app": {
+                    "guid": "a96c7067-c0cd-474a-b83b-68b1980979c2",
+                    "process": {
+                        "type": "web"
+                    }
+                },
+                "guid": "746112e9-b9e5-43a8-b48f-457da74720c0",
+                "port": 8080,
+                "selector": {
+                    "matchLabels": {
+                        "cloudfoundry.org/app_guid": "a96c7067-c0cd-474a-b83b-68b1980979c2",
+                        "cloudfoundry.org/process_type": "web"
+                    }
+                }
+            }
+        ],
+        "domain": {
+            "internal": false,
+            "name": "cf.cfi759.istio.shoot.canary.k8s-hana.ondemand.com"
+        },
+        "host": "my-app",
+        "url": "my-app.cf.cfi759.istio.shoot.canary.k8s-hana.ondemand.com"
+    }
+}
+```
+The `Istio VirtualService` is created by `RouteController` that watches the `Route CR`.
```

Comment:
I think it is worth explaining here that a new Istio Virtual service will only be created if a route has a different FQDN. So the relationship is 1 FQDN per Virtual Service and 1 Service per destination.

Reply:
Done!